### DISCUSSION TITLE
Right click node - show top 10 most used nodes to quick add?

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#10058] [Editor]
How about - Right click a node in the scene tree and it shows you a quick list of your top 10 most used nodes to quickly select and add as a child to the node you right clicked on?

Good idea or bad idea?

**Comment by Calinou** [comment]
Right-clicking in the 3D editor enables freelook, so you'd need to find another shortcut there.

**Reply by shak2** [reply]
It says "right click a node in the scene tree"...

**Reply by Calinou** [reply]
I didn't see that, apologies.

The context menu when right-clicking a node in the Scene tree dock is already pretty expansive, so I wouldn't make it even larger. While the list of most used nodes could be hidden in a submenu, this would slow down the usage of this feature somewhat as you'd have to wait for the submenu to pop up every time you create a node this way.

**Reply by shak2** [reply]
> The context menu when right-clicking a node in the Scene tree dock is already pretty expansive

True and agreed. Closed.
